auspice/client/languageSelector.js

```diff
@@ -0,0 +1,78 @@
+import React from "react"; // eslint-disable-line
+import ISO6391 from "iso-639-1";
+
+class LanguageSelector extends React.Component {
+    state = {
+      currentNarrative: undefined,
+      availableNarratives: undefined
+    }
+
+    async getAvailableNarratives() {
+      const response = await fetch("/charon/getAvailable");
+      const available = await response.json();
+      return available.narratives.map((n) => n.request);
+    }
+
+    getNarrativeLanguage(narrative) {
+      const narrativeParts = narrative.split("/");
+      let narrativeLanguage = narrativeParts[narrativeParts.length - 2];
+      if (narrativeLanguage === 'sit-rep') {
+        narrativeLanguage = 'en';
+      }
```

Comment:
There's at least 3 or 4 places where this sort of path processing is done. For maintainability, it'd be really nice to centralize it.